sklearn/utils/tests/test_array_api.py

```diff
@@ -587,11 +587,24 @@ def test_fill_or_add_to_diagonal(array_namespace, device_, dtype_name, wrap):
 
     numpy.fill_diagonal(array_np, val=1, wrap=wrap)
     with config_context(array_api_dispatch=True):
-        _fill_or_add_to_diagonal(array_xp, value=1, xp=xp, add_value=False, wrap=wrap)
+        array_xp = _fill_or_add_to_diagonal(
+            array_xp, value=1, xp=xp, add_value=False, wrap=wrap
+        )
 
     assert_array_equal(_convert_to_numpy(array_xp, xp=xp), array_np)
 
 
+def test_fill_or_add_to_diagonal_transpose():
+    """Check `_fill_or_add_to_diagonal` when `reshape` returns a copy."""
+    xp = _array_api_for_tests("numpy", None)
+    # Transposing an array makes it non-contiguous, meaning `reshape`, used within
```

Comment:
```suggestion
    # Transposing an array makes it F-contiguous, meaning `reshape(x, (-1,))`, used within
```

---

If it was `reshape(x, (-1,), order="F")`, then it would have been a view for F-contiguous arrays. For `_fill_or_add_diagonal`, I feel like we can use `np.reshape(array, (-1,), order="F")` and have a more optimal solution for NumPy and F-ordered arrays. Although, it will not be portable since the Array API spec does not have a notion of order.